src/compas/robots/model/robot.py

```diff
@@ -550,8 +565,78 @@ def get_base_link_name(self):
         >>> robot.get_base_link_name()
         'world'
         """
+        try:
+            for joint in self.iter_joint_chain():
+                if joint.is_configurable():
+                    return joint.parent.link
+        except:  # noqa: E722
+            pass
+
+        # If we cannot get the joint chain, assume ordered joints
+        # as a more or less decent default
         joints = self.get_configurable_joints()
-        return joints[0].parent.link
+        if joints and joints[0].parent:
+            return joints[0].parent.link
+
+        return None
+
+    @property
+    def rcf(self):
+        """Robot Coordinate Frame.
+
+        The RCF can be set either as a fixed joint in the model, or as an explicit frame
+        used to reposition the model. If both are used, the former takes precedence over the latter.
+
+        Returns
+        -------
+        :class:`compas.geometry.Frame`
+            Robot coordinate frame, usually placed at the base of the robot and attached to a ``world`` link.
+        """
+        return Frame.from_transformation(self._root_transformation)
+
+    @rcf.setter
+    def rcf(self, frame):
+        # RCF can be set in two ways:"
+        # 1) as a fixed frame in front of the configurable joint chain (this is the URDF way)
+        # 2) or independently of the model of joints, as a reference frame
+        # The 1) takes precedence over the 2) if both are used.
+        if not self.root or not self._joints:
+            self._rcf = frame.copy()
+            return
+
+        base_link = self.get_link_by_name(self.get_base_link_name())
+
+        # if a fixed joint exists, set its origin based on the specified frame
+        if base_link and base_link.parent_joint and base_link.parent_joint.type == Joint.FIXED:
+            base_link.parent_joint.origin = Origin(frame.point, frame.xaxis, frame.yaxis)
+        else:
+            # else, store the frame as a memory reference only
+            self._rcf = frame.copy()
+
+        self._create(self.root, self._root_transformation)
+
+    @property
+    def _root_transformation(self):
+        """Internal: root transformation used as base of the kinematic chain.
+
+        Returns
+        -------
+        :class:`compas.geometry.Transformation`
+            A transformation describing the root of the chain.
+        """
+        # Uninitialized robot, default to identity matrix
+        if not self.root or not self._joints:
+            return Transformation()
+
+        base_link = self.get_link_by_name(self.get_base_link_name())
+
+        # if there's no fixed joint and there is an RCF is explicitly set, return it as a transform
+        ref_frame = Frame.worldXY()
```

Comment:
I don't understand.  Say you've got a robot with a proper base link.  Then you set `robot.rfc = <some frame>`.  This will change the origin of the base link.  But if then you access `robot.rcf` you'll get the identity transformation in return.  Does that make sense?  Is that supposed to make sense?

Reply:
nope, the comment might be somewhat misleading if it's right above this line, but `Frame.worldXY()` is the default if the `if` right after is not true. In the case where you set the `rcf`, then the `if` goes in, and the `ref_frame` is that one instead.

Reply:
i think i didn't express myself well.  consider the following code:
```
import math
import compas_fab
from compas.robots import LocalPackageMeshLoader, RobotModel, Frame


urdf_filepath = compas_fab.get('universal_robot/ur_description/urdf/ur5.urdf')
loader = LocalPackageMeshLoader(compas_fab.get('universal_robot'), 'ur_description')
model = RobotModel.from_urdf_file(urdf_filepath)

assert model.rcf == Frame.worldXY()
model.rcf = Frame.from_euler_angles([math.pi, 0, 0], point=[1,0,0])
assert model.rcf == Frame.worldXY()
```
Without knowing the internals of `rcf`, that doesn't make any sense to the user.

Reply:
You're totally right. I'll fix.

Reply:
This should be now fixed (https://github.com/compas-dev/compas/pull/762/commits/a4a3e8f86bfd3d99c4e14f781cfb39ccebd00bab) along with simpler code path to determine what happens where (and a bit less duplication)